docs/windows-design.md

```diff
@@ -0,0 +1,255 @@
+Windows must be configured to run in the **userspace** mode, and a specific network adapter is 
+required to be provided on which the Service IP address is configured.
+
+### HNS Network configuration
+
+HNS Network is created at Antrea Agent initialization phase, and it should be created before creating 
+OVS bridge. This is because OVS is working as the Hyper-V Switch Extension, and the ovs-vswitchd process
+cannot run until the Hyper-V Switch is create and the OVS Extension is enabled on this Switch. 
+
+When creating the HNS Network, the local subnet CIDR and the uplink network adapter are required.
+Antrea Agent finds the network adapter from the Windows host using the Node internal IP as a filter,
+and retrieves the local Subnet CIDR from Node spec. 
+
+After the HNS Network is created, OVS extension should be enabled at once.
+
+### Container network configuration
+
+[**host-local**](https://github.com/containernetworking/plugins/tree/master/plugins/ipam/host-local)
+plugin is used to provide IPAM for containers, and the address is allocated from the subnet CIDR
+configured on the HNS Network. 
+
+Windows HNS Endpoint is leveraged as the vNIC for each container. A single HNSEndpoint with the
+IP allocated from IPAM plugin is created for each Pod. The HNSEndpoint should bw attached on all
+containers in the same Pod to ensure thw network configuration could be correctly accessed(this
+operation is to make sure DNS configuration is readable from all containers).
+ 
+One OVS port with the same name as the HNS Endpoint is also needed, which is helping to control
+the container traffic with OpenFlow rules. OpenFlow entries are installed to implement connectivity
+between Pod and Pod, Pod and external. The service traffic is handled by kube-proxy, and a specific
+Openflow rule is installed to output these packets to the gateway port.
+
+CNIAdd request might be called multiple times for a single Pod, this is because kubelet on Windows
+assumes CNIAdd is an idempotent event, and it uses this event to query the Pod networking status.
+Antrea needs to identify the container type(sandbox or workload) from CNIAdd request: 1) creates
+the HNSEndpoint only when the request is for the sandbox container, 2) attaches the HNSEndpoint no
+matter it is sandbox container or workload container.
+
+### Gateway port configuration
+
+Gateway port is created in Antrea Agent initialzation phase, and the address of the port should be
+the first IP in the subnet. The port is an OVS internal port and default name is "gw0".
+
+Gateway port is used to help implement L3 connectivity for the containers, including Pod to external,
+Pod to Service, and node to Pod. For Pod to external, and Pod to Service cases, openFlow entries are
+installed to output these packets to the host from the gateway port. To ensure the packet is forwarding
+correctly on the host, the IP-Forwarding feature should be enabled on the network adapter of the
+gateway port.
```

Comment:
The gateway port is used to help implement L3 connectivity for the containers, including Pod-to-external,
Pod-to-Service, and Node-to-Pod. For the Pod-to-external and Pod-to-Service cases, OpenFlow entries are
installed in order to output these packets to the host on the gateway port. To ensure the packet is forwarded
correctly on the host, the IP-Forwarding feature should be enabled on the network adapter of the
gateway port.